### PR TITLE
.gemspec tweaks

### PR DESCRIPTION
**Pull request by luke-hill**
Alter rake dependency (As mentioned in commits 10.5 is archaic, most will use a 12.x version)

Add rubocop, which is a ruby community stylistic helper tool. It gives you hints and tips to write cleaner code. I've added it into an existing code-base with the generator - So it tells you all of the mistakes or "cops offenses", but then ignores them all. This allows you to go in and slowly fix things up.

I started by fixing a load of the low hanging fruit, most of these are fairly simple to explain - Feel free to ask in review if there are any queries.

The next step would be to tackle some of the larger cop offenses and place some additional restrictions in the gemspec - But I'll save those for future PR's

### CODE REVIEW

File: test/servers_test.rb [hunk 1]
```diff
@@ -41,7 +39,7 @@ class ServersTest < Test::Unit::TestCase
             assert_equal(server_name, created_server.name)
             assert_not_nil(created_server.password)
             assert_instance_of(Array, created_server.users)
-            assert_instance_of(Fixnum, created_server.messages)
+            assert_instance_of(Integer, created_server.messages)
```
Comment (hippyhorse) [inline test/servers_test.rb]: @luke-hill Thanks for the PR, this is all good to merge except this line, which we've tried before but causes Ruby 2.2 tests to fail, as it expect Fixnum. If you can remove this we can merge the rest.

Comment (luke-hill) [reply test/servers_test.rb]: Does this mean the gem has a minimum supported version of 2.2, or lower. This wasn't evident. I'll add this into the gemspec, if you can advise @ what level it needs to be at.

**Comment by luke-hill** [comment]
I'm not sure of the failures, but given they're mentioning uninitialized class variables am I (or is there), some setup missing that I need to do. Or are we happy to merge this in as is and deal with it later?

**Comment by hippyhorse** [comment]
Closing as should be resolved by PR #27

**Pull request closed without merging**